app/helpers/patient_query_helper.rb

```diff
@@ -282,6 +282,24 @@ def advanced_filter(patients, filters, tz_offset)
         patients = advanced_filter_relative_date(patients, :symptom_onset, filter, tz_diff, :date)
       when 'continous-exposure'
         patients = patients.where(continuous_exposure: filter[:value].present? ? true : [nil, false])
+      when 'close-contact-with-known-case-id'
+        if filter[:value].blank?
+          patients = patients.where(contact_of_known_case_id: [nil, ''])
+        else
+          value_string = filter[:value].split(/\s*,\s*/).join('|')
+          case filter[:additionalFilterOption]
+          when 'Exact Match'
+            # regexp expression takes a list of strings separated by | and returns the monitorees where the
+            # contact of known case value has an exact match any of the values in the list
+            # the possible cases are: exact match of the entire value or contained in the value but preceeded and/or followed by a comma
+            # whitespace directly after the first comma and before the second is accounted for when checking possible matches
+            patients = patients.where('contact_of_known_case_id REGEXP ?', "(^|,\s?)(#{value_string})(\s?,|$)")
```

Comment:
can we also write some tests for this regex logic?

Reply:
We are in desperate need of backend tests for the advanced filter logic for sure, this would be a good start.

Reply:
here is a test file that I created that we can go off of: https://github.com/SaraAlert/SaraAlert/pull/701/commits/8ec2def47fe9a86b4822021075d1f13b09d1ce4e

Reply:
thanks @timwongj!

Reply:
You rock Tim, thanks. @sgober could you add tests for the cases to cover here and then I think this PR will be basically good to go?

Reply:
just pushed updated tests